demo-app/src/open-truss/configs/11-rest-demo.yaml

```diff
@@ -0,0 +1,33 @@
+workflow:
+  version: 1
+  signals:
+    repo_nwo_input: string
+    results:
+      total_count: number
+      incomplete_results: boolean
+  frames:
+    - frame:
+      view:
+        component: OTRestDataProvider
+        props:
+          _DEBUG_: true
+          source: rest-demo
+          path:
```

Comment:
An interesting difference between this and the other data provider is that the template is defined on the data provider whereas in the other case the template is defined in an Input component which then passes a fully formed query to the data provider.

I _think_ what you have here feels better because the result shape is defined on the data provider so it probably makes sense to also define the api call and variables here. One reason we decided to put the template in an input component was we wanted to make sure all the variables were set before we updated the query signal and triggered the API call. 

I wonder if it makes sense to add similar logic here where we don't issue the api call unless all template variables are not null. Alternatively, I wonder if it makes sense to go back to our idea of a `execute: FunctionSignal` type that components can call on demand to run the query. @andrew-dillon @kmcq @jonmagic what do you think makes the most sense from a framework design perspective?

Reply:
I like the idea of waiting to trigger the call until all the variables are set. I started implementing this but ran into some limitations with [`_.template`](https://lodash.com/docs/4.17.15#template) (it doesn't give callers access to the variables found in the parsed template string), so I'd prefer to consider this an additional feature and implement it in a new PR so as not to block this data provider (which we'll need shortly for some day job work).